lib/matrix/matrix.nit

```diff
@@ -293,3 +307,30 @@ class MatrixCoordinate
 
 	redef fun to_s do return "({x},{y})"
 end
+
+# Dedicated native structure to avoid boxing cost
+private extern class NativeMatrix `{ double* `}
+
+	new(buffer: CString) `{ return (double*)buffer; `}
```

Comment:
It's to load this structure in the Nit GC. It's very hackish as is, but I always wanted to allow FFI code to allocate space in the GC, and I see this as a trial of such a feature.